File: app/src/androidTest/java/com/mycompany/myapp/EspressoMatchers.java
package com.mycompany.myapp;

import org.hamcrest.BaseMatcher;
import org.hamcrest.Description;
import org.hamcrest.Matcher;

import java.util.regex.Pattern;

public class EspressoMatchers {
    public static Matcher<String> regex(final String regex) {
        return new BaseMatcher<String>() {
            @Override
            public boolean matches(Object item) {
                Pattern pattern = Pattern.compile(regex);
                return pattern.matcher(item.toString()).matches();
            }

            @Override
            public void describeTo(Description description) {
                description.appendText("regex '" + regex + "'");
            }
        };
    }
}

File: app/src/androidTest/java/com/mycompany/myapp/EspressoMatchers.kt
package com.mycompany.myapp

import org.hamcrest.BaseMatcher
import org.hamcrest.Description
import org.hamcrest.Matcher

import java.util.regex.Pattern

object EspressoMatchers {
    fun regex(regex: String): Matcher<String> {
        return object : BaseMatcher<String>() {
            override fun matches(item: Any): Boolean {
                val pattern = Pattern.compile(regex)
                return pattern.matcher(item.toString()).matches()
            }

            override fun describeTo(description: Description) {
                description.appendText("regex '$regex'")
            }
        }
    }
}
